Persistence/Persistence.cs

```diff
@@ -31,47 +31,50 @@ public void OnPersist(Snapshot snapshot)
-                        state["value"] = trackable.Item.ToArray().ToHexString();
-                        // Here we have a new trackable.Key and trackable.Item
-                        break;
-                    case TrackState.Changed:
-                        state["state"] = "Changed";
-                        state["key"] = trackable.Key.ToArray().ToHexString();
-                        state["value"] = trackable.Item.ToArray().ToHexString();
-                        break;
-                    case TrackState.Deleted:
-                        state["state"] = "Deleted";
-                        state["key"] = trackable.Key.ToArray().ToHexString();
-                        break;
+                    switch (trackable.State)
+                    {
+
+                        case TrackState.Added:
+                            state["state"] = "Added";
+                            state["key"] = trackable.Key.ToArray().ToHexString();
+                            state["value"] = trackable.Item.ToArray().ToHexString();
+                            // Here we have a new trackable.Key and trackable.Item
+                            break;
+                        case TrackState.Changed:
+                            state["state"] = "Changed";
+                            state["key"] = trackable.Key.ToArray().ToHexString();
+                            state["value"] = trackable.Item.ToArray().ToHexString();
+                            break;
+                        case TrackState.Deleted:
+                            state["state"] = "Deleted";
+                            state["key"] = trackable.Key.ToArray().ToHexString();
+                            break;
+                    }
+                    array.Add(state);
                 }
-                array.Add(state);
-            }
 
-            Settings.Default.BlockStorageCache = Settings.Default.BlockStorageCache + "{\"block\":" + blockIndex.ToString() + ",\"size\":" + array.Count.ToString() + ",\"storage\":\n";
-            Settings.Default.BlockStorageCache = Settings.Default.BlockStorageCache + array.ToString() + "},\n";
+                Settings.Default.BlockStorageCache = Settings.Default.BlockStorageCache + "{\"block\":" + blockIndex.ToString() + ",\"size\":" + array.Count.ToString() + ",\"storage\":\n";
+                Settings.Default.BlockStorageCache = Settings.Default.BlockStorageCache + array.ToString() + "},\n";
 
-            if ((blockIndex % Settings.Default.BlockCacheSize == 0) || (blockIndex > Settings.Default.HeightToStartRealTimeSyncing))
-            {
-                Settings.Default.BlockStorageCache += "]";
-                File.WriteAllText(path, Settings.Default.BlockStorageCache);
-                Settings.Default.BlockStorageCache = "[";
-            }   
+                if ((blockIndex % Settings.Default.BlockCacheSize == 0) || (blockIndex > Settings.Default.HeightToStartRealTimeSyncing))
```

Comment:
Good idea brother... a config variable.

Reply:
@igormcoelho, maybe put all these parameters in an array is a better idea, yep? What do you think?

Because there is a switch case at OnPersist in the way that other information can be extract if someone thing about something.